prime-router/src/main/kotlin/Element.kt

```diff
@@ -240,32 +246,41 @@ data class Element(
             }
             Type.CODE -> {
                 // First, prioritize use of a local $alt format, even if no value set exists.
-                if (format == altDisplayToken) {
-                    toAltDisplay(normalizedValue)
-                        // TODO Revisit: there may be times that normalizedValue is not an altValue
-                        ?: error("Schema Error: '$normalizedValue' is not in altValues set for '$name")
-                } else {
-                    if (valueSetRef == null)
-                        error("Schema Error: missing value set for '$name'")
-                    when (format) {
-                        caretToken -> {
-                            val display = valueSetRef.toDisplayFromCode(normalizedValue)
-                                ?: error("Internal Error: '$normalizedValue' cannot be formatted for '$name'")
-                            "$normalizedValue^$display^${valueSetRef.systemCode}"
-                        }
-                        displayToken -> {
-                            valueSetRef.toDisplayFromCode(normalizedValue)
-                                ?: error("Internal Error: '$normalizedValue' cannot be formatted for '$name'")
-                        }
-                        systemToken -> {
-                            // Very confusing, but this special case is in the HHS Guidance Confluence page
-                            if (valueSetRef.name == "hl70136" && normalizedValue == "UNK")
-                                "NULLFL"
-                            else
-                                valueSetRef.systemCode
+                when (format) {
+                    // TODO Revisit: there may be times that normalizedValue is not an altValue
+                    altDisplayToken ->
+                        toAltDisplay(normalizedValue)
+                            ?: error("Schema Error: '$normalizedValue' is not in altValues set for '$name")
+                    codeToken ->
+                        toCode(normalizedValue)
+                            ?: error(
+                                "Schema Error: '$normalizedValue' is not in valueSet '$valueSet' for '$name'/'$format'. " +
+                                    "\nAvailable values are ${valueSetRef?.values?.joinToString { "${it.code} -> ${it.display}" }}" +
+                                    "\nAlt values (${altValues?.count()}) are ${altValues?.joinToString { "${it.code} -> ${it.display}" }}"
+                            )
+                    else -> {
+                        if (valueSetRef == null)
+                            error("Schema Error: missing value set for '$name'")
+                        when (format) {
+                            caretToken -> {
```

Comment:
this when(format) else when(format) is very awkward. Can we refactor?

Reply:
Yup, I will take a look.